src/java.base/share/classes/java/net/JarURLConnection.java

```diff
@@ -152,44 +155,63 @@ public abstract class JarURLConnection extends URLConnection {
      * could be found in a specification string or the
      * string could not be parsed.
      */
-
     protected JarURLConnection(URL url) throws MalformedURLException {
         super(url);
-        parseSpecs(url);
-    }
 
-    /* get the specs for a given url out of the cache, and compute and
-     * cache them if they're not there.
-     */
-    private void parseSpecs(URL url) throws MalformedURLException {
+        // Extract JAR file URL and entry name components from the URL
         String spec = url.getFile();
+        int separatorIndex = spec.indexOf("!/");
 
-        int separator = spec.indexOf("!/");
-        /*
-         * REMIND: we don't handle nested JAR URLs
-         */
-        if (separator == -1) {
+        // REMIND: we don't handle nested JAR URLs
```

Comment:
Just noticed that this comment seems misplaced and is confusing.

The check just asserts that the URL contains the separator. I don't see how this is related to nesting at all. If the URL was nested, we would have more than one separator, right?

If we agree this comment is misplaced, perhaps we should just remove it?

Reply:
I believe this is just a reminder that there should just be one single "!/" because nested "!/" in URL are not supported.

Reply:
Alright, however mysterious it seems to carry some value. Let's keep it around.

Thanks for your reviews!

/integrate